src/uri.cr

```diff
@@ -128,7 +128,7 @@ class URI
   def full_path
     String.build do |str|
       str << (@path.try { |p| !p.empty? } ? @path : "/")
-      str << "?" << @query if @query
+      str << "?" << @query unless @query.try(&.empty?)
```

Comment:
I think this is pretty clean and eliminates the need to create an intermediate variable. I think this is a pretty small change, is reasonably easy to understand, and is unlikely to change much in the future. So it's probably ok like this. However, if a core member would like me to change it to something else though, I'm happy to do so!

Reply:
This won't work: if `@query` is `nil`, `@query.try(&.empty?)` will return `nil` which is falsey and thus add a question mark. You need add a spec for this case.

And I still think my previous proposal `(query = @query) && !query.empty?` is probably the best boolean expression for this, but maybe put it in a proper `if` branch to spare an additional ivar lookup:
```crystal
if (query = @query) && !query.empty?
  str << '?' << query
end
```

Reply:
this please

Reply:
Yeah I noticed this didn't work and updated it last night. Did you see the latest change using `try` https://github.com/crystal-lang/crystal/pull/5340/files? I think it is easier to understand, but maybe that's just me. @rx14 if you'd still prefer assigning to the intermediate variable instead of the `try` that I used, LMK and I'll change it.

Reply:
I personally think `if` branching is easier to understand over a `.try` block, but performance wise, a `try` block might show better performance than `if` branch.

https://gist.github.com/luislavena/4096693021ca38f6be4c7539625716a3

The difference might be marginal, so leave that decision to core in relation to readability/simplicity vs performance.

Cheers.